tags/value_sets.go

```diff
@@ -0,0 +1,88 @@
+package tags
+
+import (
+	"encoding/json"
+	"sort"
+)
+
+type ValueSet map[Value]struct{}
+
+func NewValueSet(vs ...Value) ValueSet {
+	rv := make(ValueSet, len(vs))
+	for _, v := range vs {
+		rv[v] = struct{}{}
+	}
+	return rv
+}
+
+// Marshals ValueSet as a sorted list.
+func (vs ValueSet) MarshalJSON() ([]byte, error) {
+	return json.Marshal(vs.AsSlice())
+}
+
+// Parses ValueSet from a list, removing duplicates.
+func (vs *ValueSet) UnmarshalJSON(b []byte) error {
+	var slice []Value
+	if err := json.Unmarshal(b, &slice); err != nil {
+		return err
+	}
+
+	*vs = NewValueSet(slice...)
+
+	return nil
+}
+
+// Adds a value to this set.
+func (vs ValueSet) Add(v Value) {
+	vs[v] = struct{}{}
+}
+
+// Adds all values from other to v.
+func (vs ValueSet) AddAll(other ValueSet) {
+	for v, _ := range other {
+		vs.Add(v)
+	}
+}
+
```

Comment:
Isn't this just `Union`? (This is called from `Tags.Union` and from the unit tests for `ValueSet`. Suggest removing the affected unit tests and changing the other call to use `Union`.)
```suggestion
```